server/mlabns/util/fqdn_rewrite.py

```diff
@@ -0,0 +1,102 @@
+"""Rewrites raw M-Lab FQDNs to apply post-processing or annotations."""
+
+import logging
+
+from mlabns.util import message
+
+
+def rewrite(fqdn, address_family, tool_id):
+    """Rewrites an FQDN to add necessary annotations and special-casing.
+
+    Performs the following rewrites on an FQDN:
+    * Adds a v4/v6 annotation if the client requested an explicit address family
+    * Applies a workaround to fix FQDNs for NDT-SSL queries.
+
+    Args:
+        fqdn: A tool FQDN with no address family specific annotation.
+        address_family: The address family for which to create the FQDN or None
+            to create an address family agnostic FQDN.
+        tool_id: Name of tool associated with the FQDN (e.g. 'ndt_ssl').
+
+    Returns:
+        FQDN after rewriting to apply all modifications to the raw FQDN.
+    """
+    rewritten_fqdn = _apply_af_awareness(fqdn, address_family)
```

Comment:
Oh, interesting. I think this is the right change for right now because the AF-annotation is just a refactoring that preserves existing behavior.

Are you sure we don't yet support IPv6? I can query mlab-ns right now and get an v6-annotated FQDN that resolves:

```
$ curl http://mlab-ns.appspot.com/npad?address_family=ipv6
{"city": "New York_NY", "url": "http://npad.iupui.mlab1v6.lga05.measurement-lab.org:8000", "ip": ["2001:1900:2100:14::12"], "fqdn": "npad.iupui.mlab1v6.lga05.measurement-lab.org", "site": "lga05", "country": "US"}

$ curl http://npad.iupui.mlab1v6.lga05.measurement-lab.org:8000
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">

<html>
<head>
  <title>NPAD Diagnostics at Measurement Lab (New York, NY, US)</title>
  <style type="text/css">
...
`